infrastructure/controller/pkg/endpoints/cluster_apply.go

```diff
@@ -0,0 +1,97 @@
+package endpoints
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/airyhq/airy/lib/go/config"
+	"github.com/airyhq/airy/lib/go/k8s"
+	"github.com/airyhq/airy/lib/go/payloads"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+type ClusterApply struct {
+	clientSet *kubernetes.Clientset
+	namespace string
+}
+
+func (s *ClusterApply) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+
+	var conf config.AiryConf
+	namespace := "default"
+	responseConfigComponents := payloads.ConfigApplyResponsePayload{}
+
+	err = json.Unmarshal(body, &conf)
+	if err != nil {
+		// TODO Remove the yaml option
+		// Legacy fallback for the airy.yaml configuration
+		yamlErr := yaml.Unmarshal(body, &conf)
+		if yamlErr != nil {
+			http.Error(w, "Error parsing the configuration.\nError:\n"+err.Error(), http.StatusBadRequest)
+			return
+		}
+	}
+
+	secData := config.GetSecurityData(conf.Security)
+	if len(secData) != 0 {
+		applyErr := k8s.ApplyConfigMap("security", namespace, secData, map[string]string{}, s.clientSet)
+		if applyErr != nil {
+			klog.Error("Unable to apply configuration for \"security\"\nError:\n" + err.Error())
```

Comment:
In this case, the request should fail and terminate early (same below)

Reply:
> In this case, the request should fail and terminate early (same below)

I am not sure about this. The components are independent from each-other, so an error in deploying one component/configMap shouldn not impact the ones that follow after. 

We can return perhaps a list of failed components

Reply:
It's true that we could do that but for clients, a completely failed state is easier to recover from than a state that is half failed. 

What do you return for a request that only did 50% of what you asked it to? `200` aka "success"? In that case, clients may not be aware that anything failed at all

Reply:
Yes, I get it :+1: We can return the `failed-components` in a separate list, but I think as it is very unlikely that only one configMap fails - I will set it to return a `500` - `InternalServerError`.
From the UI I assume it will mostly be used for applying a single component at a time anyway.